Mods request

nikifena opened the pull request:
Dear Paul. I made my version of the PCB and it works...
Here is how it looks:
https://i.postimg.cc/50CvGXLJ/IMG-20190211-133124.jpg
I decided to use 200W 19V laptop PSU which is enough for T245, but when I switch to HT420 tweezers the supply turns internal protection and stopped due to the high current consumption. 
Seems like both heaters inside the irons are getting juice from the supply simultaneous which is not good I think. So can you modify the code so only one heater to be powered at the same time?

I have another strange issue- the OLED display doesn't power on every time. I improved it by adding 10k pull-up resistors. but even that sometimes it doesn't start. I'm not sure if this an issue with the display or the code.

And one last wish for help: I swapped ENC_A and ENC_B pinouts for easy PCB arrangement but even if I change them form:
const int ENC_A            = 7;
const int ENC_B            = 6;
the encoder reduces the value moving CW not CCW.

Any suggestions?


Best Regards.
Niki

PaulusSmallus committed:
Added tweezer functionality

This includes a new PCB layout, done in KiCad. This is so it is possible to use tweezers.
The firmware has been changed to acomidate this. The interrupt part has changed a lot to allow both outputs to be individually controlled at the same time.

PaulusSmallus committed:
updated readme

Updated readme to make it clearer

PaulusSmallus committed:
change of file name

PaulusSmallus commented:
Your PCB looks really neat :) well done.

If we merge the code you want into this repository, it will no longer work with timothys circuit. What I suggest is we change the code in your branch.

My thoughts on your problems:
It probably won't be too hard to change the code so that only one heater is on at once. Although it might not heat up as quickly.

We could try adding a reset routine to the display initialization to see if that helps.

Swapping the direction of the encoder should be easy, I will just have to find it in the code.

nikifena commented:
Thanks, Paul.
This is the 3D view of the PCB: https://files.fm/u/ymgkky4h
I'm a PCB designer so know how to make PCB's. Unfortunately, my programming skills are poor.

The differences between circuits are just 4 swapped pins. It's easy to modify.

Slow heating is not a big deal. With 200W supply, the T245 heats up for 3-4 seconds. So the tweezers will do it for 10 seconds max.  

I have a spare PCB. If you or timothys want, I can give you for the tests.

PaulusSmallus commented:
That link doesn't work for me. I think you can attach files directly in the comment box if you need to.

nikifena commented:
[JBC_controller.pdf](https://github.com/timothyjager/JBC-Soldering-Controller/files/2855631/JBC_controller.pdf)
It's a 3D pdf file. You need to have Adobe Acrobat to view it. 

PS: I fixed the display issue. It was from a poor display. I tried another one and it works!
There is more. I have an issue with the second channel LED.
When I insert T245 and the iron is OFF the blue led lights.
when I turn it on the red led lights until the max temperature and then the green and the red are on.
When the iron is off and it's hot the blue and the red are on until the iron goes down to 80 degrees then the led is blue. 
I think this is the original idea but the second led has this issue:
When I insert the tweezers and when the controller is off only the first led is blue. The second is off. It must be blue too
When I turn the controller on and the tweezers are increasing the temperature, the second led turns blue (it must be red) When it reaches the maximum temperature the blue and the red are on (they must be blue and green)
When I turn off the iron and it's still hot (over 80 degrees) the blue led is on. Must be the red and blue)


Best Regards
Niki

PaulusSmallus commented:
Well done with the display.
What LEDs are you using?

nikifena commented:
Thanks.
WS2812B as shown.

PaulusSmallus committed:
added option to disable simulaneous outputs.

Added an option to disable both outputs turning on at the same time.
Set disable_simultaneous_output to true to enable.

nikifena commented:
There is more: The sleep pin /hibernation mode/ turns the heater off. I think it's better to reduce the temperature to some low value. Let's say 150 as the original JBC.
Or. If the iron is on the stand the temperature goes down to 150C and if it is more than 10 minutes this will turn off the iron or goes down to 60-70 degrees.


And there is more:
When I short the sleep in and ground and then open it, the green LEDs lights when the iron heats up instead of the red one. There is another small issue there I think...

nikifena commented:
![scr04](https://user-images.githubusercontent.com/45510804/52642433-adcf2b00-2ee3-11e9-98da-d8bc13022c56.PNG)
Seems like the main channel is not powered the same way as the other one. and finally, it takes a little bit more time to heat up the other iron.

The scope print shows the heater traces for both channels

PaulusSmallus committed:
update readme
